src/data/bucket/fill_extrusion_bucket.js

```diff
@@ -127,28 +127,21 @@ class FillExtrusionBucket implements Bucket {
                 numVertices += ring.length;
             }
 
-            const segment = this.segments.prepareSegment(numVertices * 5, this.layoutVertexArray, this.indexArray);
-
-            const flattened = [];
-            const holeIndices = [];
-            const indices = [];
+            let segment = this.segments.prepareSegment(4, this.layoutVertexArray, this.indexArray);
 
             for (const ring of polygon) {
                 if (ring.length === 0) {
                     continue;
                 }
 
-                if (ring !== polygon[0]) {
-                    holeIndices.push(flattened.length / 2);
-                }
-
                 let edgeDistance = 0;
 
                 for (let p = 0; p < ring.length; p++) {
-                    const p1 = ring[p];
+                    if (segment.vertexLength + 4 > MAX_VERTEX_ARRAY_LENGTH) {
```

Comment:
Should this be inside the `p >= 1` and `!isBoundaryEdge(p1, p2)` conditionals, so we only prepare a new segment if's definitely needed?